### PR TITLE
Add a helm chart for openfunction and its dependencies

### PR DESCRIPTION
**Pull request by wrongerror**
Add a helm chart for openfunction and its dependencies that do not officially support helm deployments, currently it only supports kubeVersion 1.20+

Signed-off-by: wrongerror <wangyifei@kubesphere.io>

### CODE REVIEW

File: charts/knative-serving/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+# to be deployed.
+#
+# Library charts provide useful utilities or functions for the chart developer. They're included as
+# a dependency of application charts to inject those utilities and functions into the rendering
+# pipeline. Library charts do not define any templates and therefore cannot be deployed.
+type: application
+# This is the chart version. This version number should be incremented each time you make changes
+# to the chart and its templates, including the app version.
+# Versions are expected to follow Semantic Versioning (https://semver.org/)
+version: 1.0.1
```
Comment (benjaminhuo) [inline charts/knative-serving/Chart.yaml:16]: The chart's version shouldn't be the same as appVersion.
The chart version can be set to v0.1.0

File: charts/knative-serving/templates/configmaps/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  name: config-deployment
+  labels:
+    app.kubernetes.io/part-of: knative-serving
+    app.kubernetes.io/version: 1.0.1
+    serving.knative.dev/release: v1.0.1
+  {{- include "knative-serving.labels" . | nindent 4 }}
+  annotations:
+    knative.dev/example-checksum: dd7ee769
+data:
+  _example: {{ .Values.configDeployment.Example | quote }}
```
Comment (benjaminhuo) [inline charts/knative-serving/templates/configmaps/deployment.yaml:13]: The `_example` field is just for demonstration purposes as I can remember @tpiperatgod, right?. Maybe we needn't set it to a helm value.

File: charts/knative-serving/Chart.yaml (hunk 1 above)
Comment (wrongerror) [reply charts/knative-serving/Chart.yaml:16]: I also set verison to 0.1.0 before, but considering the possibility of supporting lower versions of the component later, I modified version to adapt to appVersion

File: charts/knative-serving/templates/configmaps/deployment.yaml (hunk 2 above)
Comment (tpiperatgod) [reply charts/knative-serving/templates/configmaps/deployment.yaml:13]: yes, it's just for demonstration

File: charts/knative-serving/Chart.yaml (hunk 1 above)
Comment (benjaminhuo) [reply charts/knative-serving/Chart.yaml:16]: I don't understand.
version is helm chart's version
The helm chart install will support knative v1.0.1+ starting from k8s 1.20, lower version k8s doesn't have helm install method.

image

File: charts/knative-serving/templates/configmaps/deployment.yaml (hunk 2 above)
Comment (benjaminhuo) [reply charts/knative-serving/templates/configmaps/deployment.yaml:13]: Then we needn't to turn the _example into a value

Comment (wrongerror) [reply charts/knative-serving/templates/configmaps/deployment.yaml:13]: already move to template file

File: charts/openfunction/values.yaml [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    plugins: |
+      pre:
+      - plugin1
+      - plugin2
+      post:
+      - plugin2
+      - plugin1
+    plugins.tracing: |
+      # Switch for tracing, default to false
+      enabled: true
```
Comment (benjaminhuo) [inline charts/openfunction/values.yaml:15]: default should be false, not ture

Comment (wrongerror) [reply charts/openfunction/values.yaml:15]: https://github.com/OpenFunction/OpenFunction/blob/main/config/bundle.yaml#L15292

Comment (benjaminhuo) [reply charts/openfunction/values.yaml:15]: The original is not correct, correct it together

**Comment by wrongerror** [comment]
Since we decided to move the helm chart related content to the new repo: https://github.com/OpenFunction/charts, this pr is no longer needed

**Pull request closed without merging**